### PR TITLE
D8CORE-2544: added (link is external) to the brandbar

### PR DESCRIPTION
**Pull request by cjwest**
# READY FOR REVIEW

# Summary
Adds  `<span class="su-brand-bar__link--external">(link is external)</span>` to the external link in the brand bar
This will announce to a screen reader that this is an external link.

# Needed By (Date)
- Soon

# Urgency
- Not urgent

# Steps to Test

1. Check out this branch in Stanford Basic
2. run npm install in Stanford Basic
3. Run npm run publish to populate /decanter in the Stanford Basic templates directory
4. Inspect the brand bar to verify `<span class="su-brand-bar__link--external">(link is external)</span>`accompanies the external link in the global header.

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
- JIRA ticket https://stanfordits.atlassian.net/browse/D8CORE-2544
- Related PRs: https://github.com/SU-SWS/decanter/pull/716
- Any other contextual information that might be helpful (e.g., description of a bug that this PR fixes, new functionality that it adds, etc.)
- Anyone who should be notified? (`@mention` them here)
@JBCSU 

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

File: core/dist/css/decanter-grid.css [hunk 1]
```diff
@@ -1 +1,1909 @@
...h:calc(41.66667% - 28px)}}@media only screen and (min-width:1500px){.flex-2xl-6-of-12{-webkit-box-flex:0;-ms-flex:0 0 calc(50% - 24px);flex:0 0 calc(50% - 24px);max-width:calc(50% - 24px)}}@media only screen and (min-width:1500px){.flex-2xl-7-of-12{-webkit-box-flex:0;-ms-flex:0 0 calc(58.33333% - 20px);flex:0 0 calc(58.33333% - 20px);max-width:calc(58.33333% - 20px)}}@media only screen and (min-width:1500px){.flex-2xl-8-of-12{-webkit-box-flex:0;-ms-flex:0 0 calc(66.66667% - 16px);flex:0 0 calc(66.66667% - 16px);max-width:calc(66.66667% - 16px)}}@media only screen and (min-width:1500px){.flex-2xl-9-of-12{-webkit-box-flex:0;-ms-flex:0 0 calc(75% - 12px);flex:0 0 calc(75% - 12px);max-width:calc(75% - 12px)}}@media only screen and (min-width:1500px){.flex-2xl-10-of-12{-webkit-box-flex:0;-ms-flex:0 0 calc(83.33333% - 8px);flex:0 0 calc(83.33333% - 8px);max-width:calc(83.33333% - 8px)}}@media only screen and (min-width:1500px){.flex-2xl-11-of-12{-webkit-box-flex:0;-ms-flex:0 0 calc(91.66667% - 4px);flex:0 0 calc(91.66667% - 4px);max-width:calc(91.66667% - 4px)}}@media only screen and (min-width:1500px){.flex-2xl-12-of-12{-webkit-box-flex:0;-ms-flex:0 0 100%;flex:0 0 100%;max-width:100%}}
```
Comment (sherakama) [inline core/dist/css/decanter-grid.css:1]: @cjwest please compile with `npm run publish`

File: core/src/templates/components/brand-bar/brand-bar.twig [hunk 2]
```diff
@@ -14,6 +14,12 @@
  * - modifier_class: Additional css classes to change look and behaviour.
  */
 #}
+{% if external_link_text is empty -%}
+  {%- set external_link_text -%}
+    <span class="su-brand-bar__link--external">(link is external)</span>
```
Comment (sherakama) [inline core/src/templates/components/brand-bar/brand-bar.twig:19]: Strings must be translated.
```suggestion
    <span class="su-brand-bar__link--external">({{ "link is external"|t }})</span>
```

Comment (sherakama) [reply core/src/templates/components/brand-bar/brand-bar.twig:19]: Are the round brackets neccessary? "()"

Comment (sherakama) [reply core/src/templates/components/brand-bar/brand-bar.twig:19]: According to BEM `--` is used for a modifier to an existing element. This is a whole new element. Perhaps name it `su-brand-bar__link-a11y`?

File: core/src/templates/components/brand-bar/brand-bar.twig [hunk 3]
```diff
@@ -14,6 +14,12 @@
...
+  {% endset %}
+{%- endif %}
+
 <div class="su-brand-bar {{ modifier_class }}">
-  <div class="su-brand-bar__container"><a {{ attributes }} class="su-brand-bar__logo" href="https://stanford.edu">Stanford University</a></div>
+  <div class="su-brand-bar__container"><a {{ attributes }} class="su-brand-bar__logo" href="https://stanford.edu">Stanford University {{ external_link_text }}</a></div>
```
Comment (sherakama) [inline core/src/templates/components/brand-bar/brand-bar.twig:24]: Links will include this extra space now. I'm not sure it matters here but it should be called out.

File: core/src/templates/components/brand-bar/brand-bar.twig (hunk 2 above)
Comment (sherakama) [reply core/src/templates/components/brand-bar/brand-bar.twig:19]: Ignore the "strings must be translated" comment. That is a Drupal thing, and not a Decanter thing.

**Comment by sherakama** [comment]
@cjwest feedback provided.

File: core/src/templates/components/brand-bar/brand-bar.twig (hunk 2 above)
Comment (cjwest) [reply core/src/templates/components/brand-bar/brand-bar.twig:19]: @sherakama ,
With regard to the round brackets (often known as parenthesis), I'm seeing this implemented about half the time. Parenthesis usually indicate that the information within is tangental, and I kind of like that convention. 
If you take this as an example:

I find this harder to read:
<a href="1040.pdf">Tax form 1041 PDF</a>

Than this:
<a href="1040.pdf">Tax form 1041 (PDF)</a>

As we move forward with other icons, I'd like to keep the round brackets "()" as a convention to indicate that the information about the icon is tangental.

Comment (jbickar) [reply core/src/templates/components/brand-bar/brand-bar.twig:19]: > round brackets (often known as parenthesis)

Lulz

**Comment by sherakama** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr717-oqjqueerytyujlb7ehp8b19vmjdeyqhu.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5f46b19b6bc75a79470bac58

**Comment by sherakama** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr717-oqjqueerytyujlb7ehp8b19vmjdeyqhu.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5f46b19b6bc75a79470bac58

File: core/src/templates/components/brand-bar/brand-bar.twig (hunk 2 above)
Comment (sherakama) [reply core/src/templates/components/brand-bar/brand-bar.twig]: Screen Shot 2020-08-27 at 1 54 16 PM

**Pull request merged**